$OpenBSD: patch-netwerk_ipc_SocketProcessImpl_cpp,v 1.1 2021/07/13 15:25:30 landry Exp $

https://bugzilla.mozilla.org/show_bug.cgi?id=1713999
https://phabricator.services.mozilla.com/D116641
to test at runtime:
- run with MOZ_LOG=socketprocess:5
- set MOZ_FORCE_USE_SOCKET_PROCESS=1 in the env
- and/or network.process.enabled=1
- and/or network.http.network_access_on_socket_process.enabled=1
cf https://searchfox.org/mozilla-central/source/netwerk/base/nsIOService.cpp#588

Index: netwerk/ipc/SocketProcessImpl.cpp
--- netwerk/ipc/SocketProcessImpl.cpp.orig
+++ netwerk/ipc/SocketProcessImpl.cpp
@@ -16,6 +16,9 @@
 
 #if defined(OS_WIN) && defined(MOZ_SANDBOX)
 #  include "mozilla/sandboxTarget.h"
+#elif defined(__OpenBSD__) && defined(MOZ_SANDBOX)
+#  include "mozilla/SandboxSettings.h"
+#  include "prlink.h"
 #endif
 
 #ifdef OS_POSIX
@@ -47,6 +50,11 @@ bool SocketProcessImpl::Init(int aArgc, char* aArgv[])
   LoadLibraryW(L"softokn3.dll");
   LoadLibraryW(L"freebl3.dll");
   mozilla::SandboxTarget::Instance()->StartSandbox();
+#elif defined(__OpenBSD__) && defined(MOZ_SANDBOX)
+  PR_LoadLibrary("libnss3.so");
+  PR_LoadLibrary("libsoftokn3.so");
+  PR_LoadLibrary("libfreebl3.so");
+  StartOpenBSDSandbox(GeckoProcessType_Socket);
 #endif
   char* parentBuildID = nullptr;
   char* prefsHandle = nullptr;
